dask_sql/physical/rel/logical/aggregate.py

```diff
@@ -231,6 +231,15 @@ def _do_aggregations(
         for col in group_columns:
             collected_aggregations[None].append((col, col, "first"))
 
+        groupby_agg_options = context.schema[
+            context.schema_name
+        ].config.get_config_by_prefix("dask.groupby.aggregate")
+        # Update the config string to only include the actual param value
+        # i.e. dask.groupby.aggregate.split_out -> split_out
```

Comment:
```suggestion
```

Reply:
Thanks. I had initially planned on having it as a part of the config methods but then ran into the issue where this might be problematic in situations where different config options have the same suffix string [see](https://github.com/dask-contrib/dask-sql/pull/286#discussion_r743322304).

I wasn't able to think of a clever way to avoid that and ended up adding the logic to the groupby bit only but am open to suggestions.  
I wonder if it's worth adding some sort of `strip_prefix` utility method to the config that can be called by different users of the config options. It has the added benefit of being able to check if the `key` already exists in the values returned and warn. So if two config options map to the same key after getting rid of the prefix, it could potentially warn/raise.

Reply:
Ah that makes sense - I was assuming that in most cases, we would use `get_config_by_prefix` with a relatively nested prefix - so something like `dask.groupby.aggregate` instead of `dask`.

Maybe we could return a nested dictionary for `get_config_by_prefix`? Not really sure of an efficient way to do this right now, but was thinking something like

```python
sql_config = ConfigContainer()
sql_config.set_config(
    {
        "dask.groupby.aggregate.split_out":1,
        "dask.groupby.aggregate.split_every": 1,
        "dask.dataframe.drop_duplicates.split_out": 3,
    }
)

sql_config.get_config_by_prefix("dask")
{
    'groupby': {
        'aggregate': {
            'split_out': 1,
            'split_every': 1
        }
    },
    'dataframe': {
        'drop_duplicates': {
            'split_out': 3
        }
    }
}
```

Reply:
Yeah, that makes sense and it's one of the reasons that prompted me to consider using Dask's config module that implements configurations as nested dictionaries while still allowing nested access using the `.` operator with [get](https://docs.dask.org/en/stable/configuration.html#api).
It also has the added benefit of allowing configurations to be set in yaml files in case users prefer to set configs globally before execution. 
I'm guessing all of this discussion is leading me to believe that having nested dicts of dicts as the config and using Dask's config implementation seems to be a cleaner approach to handle this. 
@charlesbluca Curious to hear what your thoughts are

Reply:
Yeah, I think using dask's config module makes sense contextually here and might make it easier to consolidate things in the long term.

Not sure if that entails a large overhaul on this PR; if that's the case then I'm okay with merging this in as is (since groupby agg is really the only place we use the config container currently), and doing that work in a separate follow up PR, ideally as we find more place where would want to inject Dask-level configuration.